packages/cardano-services/test/StakePool/TypeormStakePoolProvider/TypeormStakePoolProvider.test.ts

```diff
@@ -0,0 +1,750 @@
+            } catch (error: any) {
+              expect(error.response.status).toBe(400);
+              expect(error.message).toBe(BAD_REQUEST);
+            }
+          });
+
+          it('returns a 400 coded error if pagination limit is greater than pagination page size limit', async () => {
+            expect.assertions(2);
+            const pageSizeGreaterThanMaxLimit = 30;
+            try {
+              await axios.post(
+                `${baseUrl}${url}`,
+                { pagination: { limit: pageSizeGreaterThanMaxLimit, startAt: 0 } },
+                { headers: { 'Content-Type': APPLICATION_JSON } }
+              );
+            } catch (error: any) {
+              expect(error.response.status).toBe(400);
+              expect(error.message).toBe(BAD_REQUEST);
+            }
+          });
+
+          it('returns a 400 coded error if provided filter identifier values are greater than pagination page size limit', async () => {
+            expect.assertions(2);
+            const filters = {
+              identifier: {
+                _condition: 'or',
+                values: [
+                  { name: 'Test1' },
+                  { name: 'Test2' },
+                  { ticker: 'Test3' },
+                  { ticker: 'Test4' },
+                  { ticker: 'Test5' },
+                  { ticker: 'Test6' },
+                  { ticker: 'Test7' },
+                  { ticker: 'Test8' },
+                  { ticker: 'Test9' },
+                  { id: 'pool1jcwn98a6rqr7a7yakanm5sz6asx9gfjsr343mus0tsye23wmg70' as Cardano.PoolId },
+                  { id: '98a6rqr7a7yakanm5sz6asx9gfjsr343mus0tsye23wmg70' as Cardano.PoolId }
+                ]
+              }
+            };
+
+            try {
+              await axios.post(
+                `${baseUrl}${url}`,
+                { filters, pagination: { limit: 5, startAt: 0 } },
+                { headers: { 'Content-Type': APPLICATION_JSON } }
+              );
+            } catch (error: any) {
+              expect(error.response.status).toBe(400);
+              expect(error.message).toBe(BAD_REQUEST);
+            }
+          });
+        });
+
+        it('response is an array of stake pools', async () => {
+          const options: QueryStakePoolsArgs = {
+            filters: {
+              identifier: {
+                _condition: 'or',
+                values: [{ id: poolsInfo[0].id }, { id: poolsInfo[1].id }]
+              }
+            },
+            pagination
+          };
+          const response = await provider.queryStakePools(options);
+
+          expect(() => Cardano.PoolId(poolsInfo[0].id as unknown as string)).not.toThrow();
+          expect(() => Cardano.PoolIdHex(response.pageResults[0].hexId as unknown as string)).not.toThrow();
+          expect(() => Cardano.VrfVkHex(response.pageResults[0].vrf as unknown as string)).not.toThrow();
```

Comment:
Would be great to add more assertions like that to ensure we're returning an object that matches StakePool type (including metrics & metadata)